### ISSUE TITLE
Get RBG image from NV21 frame

### ISSUE THREAD

**Issue by Osarez7** [saki4510t/UVCCamera#191]
Hi , I'm working in an app and I need to get two types of images from the camera frame, one is a byte array in NV21 format and the other one is a  RGB bitmap

I'm using the IFrameCallback to get the frame from the camera
`mUVCCamera.setFrameCallback(mIFrameCallback, UVCCamera.PIXEL_FORMAT_NV21);`

Then I get the byte array that I need 
```java
byte[] mFrame;
private final IFrameCallback mIFrameCallback = new IFrameCallback() {
    @Override
    public void onFrame(final ByteBuffer frameBuffer) {
           mFrame = new byte[frameBuffer.remaining()];
           frameBuffer.get(mFrame, 0, mFrame.length);
     }
}
```
In other method I get the bitmap 

```java
public Bitmap getBitmap(){
    if (mFrame != null) {
                YuvImage yuv = new YuvImage(mFrame, ImageFormat.NV21, mFrameWidth, mFrameHeight, null);
                ByteArrayOutputStream os = new ByteArrayOutputStream();
                yuv.compressToJpeg(new Rect(0, 0, mFrameWidth, mFrameHeight), 100, os);
                byte[] bytes = os.toByteArray();
                return  BitmapFactory.decodeByteArray(bytes, 0, bytes.length);
            }
}
```

The bitmap that I'm getting has a wrong format, for example in this image the object has to be orange instead of blue

![image](https://user-images.githubusercontent.com/1088494/27886343-480af3a0-61a1-11e7-9e1c-0e5812c291b4.png)

Any advice?

**Comment by FanHongchuang** [comment]
@Osarez7 
Hello,
Are you resolve it?

**Comment by gaborbencebekesi** [comment]
Hello @Osarez7,

Did you solve this problem?

Thanks,
Gabor

**Comment by capsane** [comment]
hi, when I use PIXEL_FORMAT_NV21, the format of the picture was wrong, which swapped the red and blue channel.
In the end, I change the format into PIXEL_FORMAT_YUV420SP like this:
`mUVCCamera.setFrameCallback(callback, UVCCamera.PIXEL_FORMAT_YUV420SP);`
then I get the rigth picture.
btw, try using RenderScript and ScriptIntrinsicYuvToRGB to accelerate the transforming.

**Comment by janoonaj** [comment]
Yes, there is a problem with the enums on the library. If you want N21 you need to write YUV420SP

More here: [https://github.com/saki4510t/UVCCamera/issues/207](https://github.com/saki4510t/UVCCamera/issues/207)

**Comment by liqvip** [comment]
@Osarez7 Your code showed me how to convert ByteBuffer to a byte array, thank you very much!
